test/e2e/set/scylladbcluster/multidatacenter/scylladbcluster_fault_tolerance.go

```diff
@@ -0,0 +1,184 @@
+package multidatacenter
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	"github.com/scylladb/scylla-operator/pkg/pointer"
+	scyllafixture "github.com/scylladb/scylla-operator/test/e2e/fixture/scylla"
+	"github.com/scylladb/scylla-operator/test/e2e/framework"
+	"github.com/scylladb/scylla-operator/test/e2e/utils"
+	v1alpha1utils "github.com/scylladb/scylla-operator/test/e2e/utils/v1alpha1"
+	scylladbclusterverification "github.com/scylladb/scylla-operator/test/e2e/utils/verification/scylladbcluster"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = g.Describe("Multi datacenter ScyllaDBCluster", framework.MultiDatacenter, func() {
+	f := framework.NewFramework("scylladbcluster")
+
+	g.It("should reconcile healthy datacenters when any of DCs are down", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		rkcs, rkcClusterMap, err := utils.SetUpRemoteKubernetesClustersFromRestConfigs(ctx, framework.TestContext.RestConfigs, f)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		o.Expect(len(rkcs)).To(o.BeNumerically(">=", 2))
+
+		framework.By("Creating ScyllaDBCluster")
+		scyllaConfigCM := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "scylla-config",
+			},
+			Data: map[string]string{
+				"scylla.yaml": `read_request_timeout_in_ms: 1234`,
+			},
+		}
+
+		const agentAuthToken = "foobar"
+		scyllaManagerAgentConfigSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "scylla-manager-agent-config",
+			},
+			StringData: map[string]string{
+				"scylla-manager-agent.yaml": fmt.Sprintf(`auth_token: %s`, agentAuthToken),
+			},
+		}
+
+		scyllaConfigCM, err = f.KubeClient().CoreV1().ConfigMaps(f.Namespace()).Create(ctx, scyllaConfigCM, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		scyllaManagerAgentConfigSecret, err = f.KubeClient().CoreV1().Secrets(f.Namespace()).Create(ctx, scyllaManagerAgentConfigSecret, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		sc := f.GetDefaultScyllaDBCluster(rkcs)
+		sc.Spec.DatacenterTemplate.ScyllaDB.CustomConfigMapRef = pointer.Ptr(scyllaConfigCM.Name)
+		sc.Spec.DatacenterTemplate.ScyllaDBManagerAgent = &scyllav1alpha1.ScyllaDBManagerAgentTemplate{
+			CustomConfigSecretRef: pointer.Ptr(scyllaManagerAgentConfigSecret.Name),
+		}
+
+		metaCluster := f.Cluster(0)
+		sc, err = f.ScyllaClient().ScyllaV1alpha1().ScyllaDBClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
```

Comment:
nit: for clarity, shouldn't this use `metaCluster` instead of `f` directly as well?